src/LabelStudio.js

```diff
@@ -62,20 +64,68 @@ export class LabelStudio {
     this.store = store;
     window.Htx = this.store;
 
-    render((
-      <App
-        store={this.store}
-        panels={registerPanels(this.options.panels) ?? []}
-      />
-    ), rootElement);
+    const isRendered = false;
+
+    const renderApp = () => {
+      if (isRendered) {
+        clearRenderedApp();
+      }
+      render((
+        <App
+          store={this.store}
+          panels={registerPanels(this.options.panels) ?? []}
+        />
+      ), rootElement);
+    };
+
+    const clearRenderedApp = () => {
+      const childNodes = [...rootElement.childNodes];
+      // cleanDomAfterReact needs this key to be sure that cleaning affects only current react subtree
+      const reactKey = findReactKey(childNodes[0]);
 
-    const destructor = () => {
       unmountComponentAtNode(rootElement);
+      /*
+        Unmounting doesn't help with clearing React's fibers
+        but removing the manually helps
+        @see https://github.com/facebook/react/pull/20290 (similar problem)
+        That's maybe not relevant in version 18
```

Comment:
Yea this will not be a problem once we move to React 18, and use the same compiled React throughout. In fact, the latter is likely what is snagging this now in React 17. Because we are operating with separate compiled React versions (even though they are the same version) it is not going to be referencing the identical code points, just an adjacent *similar* code point. That can only cause problems within the context of React's operations.